### PR TITLE
A handful of guncode tweaks

### PR DESCRIPTION
**Pull request by Tarmunora**
## About the PR 
Makes a few tweaks to guns that I felt were cool/needed/missing:
Burstfire guns not burstfiring on pointblank was weird and dumb, so fixed
click a magazine with another one (of the same type) to transfer ammo
adds some projectile spread to guns while drunk - enough to be noticeable , not enough to be really devastating hopefully
adds support for giving guns different rates of fire - doesn't actually change any, that can be considered for future changes... except on assday, when it defaults to 0 ticks of delay. heh.

## Why's this needed? 
A bit of bugfixing, a bit of QoL changes, and the image of the drunken det shooting a *little* more wildly than normal is really funny, and rate of fire stuff gives another knob to tweak for gun balance

## Changelog

```
Tarmunora:
* Guns can now have a custom click delay (rate of fire). Delay is currently kept at 4 for all guns (no change), except on ass day, where you can shoot as fast as you can click.
* Guns have a little bit of extra spread when drunk. Alcohol immunity mitigates this, to an extent
* Added the ability to refill/combine ammo boxes by hitting them with a box of the same type
* Changed pointblanking to hit with all the projectiles in a burst, not just one
* Fixes up gunpoint grabs a little bit (should shoot more consistently)
```

### CODE REVIEW

**Comment by Gannets** [comment]
Without looking at the code, these all sound like sensible changes!

**Comment by Kyle2143** [comment]
I like how the more drunk you are the more spread there is, but I'm concerned that the spread numbers are too high. Max spread looks like it would be 15 degrees and that's at >110 units of ethanol, which to be fair is very drunk, but still that's like 2 sips from Bojack Daniels for the detective.. IIRC 15 degrees is a considerable spread, such that it's almost unable to hit anything more than 2 tiles away, I might be wrong, but if so that sounds too bad. I like how the drunkenness values you chose correspond to the drunkenness status effect.

**Comment by Tarmunora** [comment]
> 
> 
> I like how the more drunk you are the more spread there is, but I'm concerned that the spread numbers are too high. Max spread looks like it would be 15 degrees and that's at >110 units of ethanol, which to be fair is very drunk, but still that's like 2 sips from Bojack Daniels for the detective.. IIRC 15 degrees is a considerable spread, such that it's almost unable to hit anything more than 2 tiles away, I might be wrong, but if so that sounds too bad. I like how the drunkenness values you chose correspond to the drunkenness status effect.

It caps out at 10 degrees spread, and alcohol resistance (which det/barman/HoS start the round with) reduces the severity, so they get no effect until 110u, at which point they cap out at 5 degree spread.

File: code/obj/item/grab.dm [hunk 1]
```diff
@@ -660,9 +660,8 @@
 		shot = 1
 
 		if (affecting && assailant && isitem(src.loc))
-			if (get_dist(src.affecting,src.assailant) <= 1)
```
Comment (ZeWaka) [inline code/obj/item/grab.dm:663]: Why is this removed? This seems necessary.

Comment (Tarmunora) [reply code/obj/item/grab.dm:663]: Distance check is moved to the pointblanking code - it fires a bullet at the target instead of pointblanking if they are more than 1 tile distant. Also, I just realized I need to fix something else. DNM until I get another commit in please

Comment (Tarmunora) [reply code/obj/item/grab.dm:663]: Ok, what I thought was an issue (on_pointblank calls) is actually handled in projectile_parent and I don't need to worry about it. nevermind.

**Comment by Adharainspace** [comment]
i think being drunk causing spread makes sense, since being drunk buffs your melee damage doesnt it? maybe only fists but still
